src/test/update-course-links.test.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+const path = require('path');
+const test = require('ava');
+const fetchMock = require('fetch-mock');
+const parser = require('parse5');
+
+const CourseLinkUpdater = require('../update-course-links');
+
+const ContentPath = path.join(__dirname, 'content');
+const ManifestPath = path.join(__dirname, 'content/manifest.json');
+
+const MockValence = {
+	createAuthenticatedUrl(url) {
+		return url;
+	}
+};
+
+const OrgUnit = {
+	Identifier: '123',
+	Name: 'Org Unit',
+	Path: '/content/course123/'
+};
+
+test('_findInManifest finds topic in module', t => {
+	const manifest = {
+		modules: [{
+			title: 'module1',
+			children: [{
+				title: 'topic1'
+			}]
+		}]
+	};
+
+	class MockModuleProcessor { }
+
+	const fetch = fetchMock.sandbox();
+
+	const processor = new CourseLinkUpdater({ contentDirectory: ContentPath, manifestPath: ManifestPath }, MockValence, fetch, MockModuleProcessor);
+
+	const result = processor._findInManifest(x => x.title === 'topic1', manifest);
+
+	t.deepEqual(result, {
+		title: 'topic1'
+	});
+});
+
+test('_findInManifest finds topic in submodule', t => {
+	const manifest = {
+		modules: [{
+			title: 'module1',
+			children: [{
+				title: 'module2',
+				children: [{
+					title: 'topic1'
+				}]
+			}]
+		}]
+	};
+
+	class MockModuleProcessor { }
+
+	const fetch = fetchMock.sandbox();
+
+	const processor = new CourseLinkUpdater({ contentDirectory: ContentPath, manifestPath: ManifestPath }, MockValence, fetch, MockModuleProcessor);
+
+	const result = processor._findInManifest(x => x.title === 'topic1', manifest);
+
+	t.deepEqual(result, {
+		title: 'topic1'
+	});
+});
+
+test('_findInManifest finds module in manifest', t => {
+	const manifest = {
+		modules: [{
+			title: 'module1'
+		}]
+	};
+
+	class MockModuleProcessor { }
+
+	const fetch = fetchMock.sandbox();
+
+	const processor = new CourseLinkUpdater({ contentDirectory: ContentPath, manifestPath: ManifestPath }, MockValence, fetch, MockModuleProcessor);
+
+	const result = processor._findInManifest(x => x.title === 'module1', manifest);
+
+	t.deepEqual(result, {
+		title: 'module1'
+	});
+});
+
+test('_getAllLinks returns all links in document', t => {
+	const document = '<html><body><p>Text <a href="link.html">Link</a></p></body></html';
+
+	class MockModuleProcessor { }
+
+	const fetch = fetchMock.sandbox();
+
+	const processor = new CourseLinkUpdater({ contentDirectory: ContentPath, manifestPath: ManifestPath }, MockValence, fetch, MockModuleProcessor);
+
+	const links = processor._getAllLinks(parser.parse(document));
+
+	for (const link of links) {
+		t.truthy(link.attrs.find(x => x.name === 'href'));
+	}
+});
+
+test('updateLinks updates the links', async t => {
```

Comment:
Is this really testing anything?   It's just asserting that the manifest was read and passed into processModule, isn't it?